### PR TITLE
Collateral validation

### PR DESCRIPTION
**Pull request by sypeer**
TLRV2-426

### CODE REVIEW

File: packages/contracts/contracts/TellerV2.sol [hunk 1]
```diff
@@ -244,6 +247,47 @@ contract TellerV2 is
...
+        uint256 _principal,
+        uint32 _duration,
+        uint16 _APR,
+        string calldata _metadataURI,
+        address _receiver,
+        address _collateralAddress,
+        ICollateralEscrowFactory.Collateral calldata _collateralInfo
+    ) public override whenNotPaused returns (uint256 bidId_) {
+        bidId_ = _submitBid(_lendingToken, _marketplaceId, _principal, _duration, _APR, _metadataURI, _receiver);
+        bool validation = collateralEscrowFactoryBeacon.validateCollateral(bidId_, _msgSenderForMarket(_marketplaceId), _collateralAddress, _collateralInfo);
```
Comment (passabilities) [inline packages/contracts/contracts/TellerV2.sol:277]: lets use the stored `borrower` for the submitted bid
```suggestion
        bool validation = collateralEscrowFactoryBeacon.validateCollateral(bidId_, bids[bidId_].borrower, _collateralAddress, _collateralInfo);
```

File: packages/contracts/contracts/TellerV2MarketForwarder.sol [hunk 2]
```diff
@@ -78,9 +78,13 @@ abstract contract TellerV2MarketForwarder is Initializable, ContextUpgradeable {
...
         bytes memory responseData;
 
+        bytes4 submitBidSelector = bytes4(
+            keccak256("ITellerV2.submitBid(address,uint256,uint256,uint32,uint16,string,address)")
+        );
+
         responseData = _forwardCall(
             abi.encodeWithSelector(
-                ITellerV2.submitBid.selector,
+                submitBidSelector,
```
Comment (passabilities) [inline packages/contracts/contracts/TellerV2MarketForwarder.sol:87]: we should be able to simplify this to:
```suggestion
        responseData = _forwardCall(
            abi.encodeWithSignature(
                "submitBid(address,uint256,uint256,uint32,uint16,string,address)",
```

File: packages/contracts/contracts/TellerV2Storage.sol [hunk 3]
```diff
@@ -138,4 +139,9 @@ abstract contract TellerV2Storage_G2 is TellerV2Storage_G1 {
     address public lenderCommitmentForwarder;
 }
 
-abstract contract TellerV2Storage is TellerV2Storage_G2 {}
+abstract contract TellerV2Storage_G3 is TellerV2Storage_G2 {
+    ICollateralEscrowFactory public collateralEscrowFactoryBeacon;
```
Comment (passabilities) [inline packages/contracts/contracts/TellerV2Storage.sol:143]: this contract isn't the actual beacon so lets just name it as the factory
```suggestion
    ICollateralEscrowFactory public collateralEscrowFactory;
```

File: packages/contracts/contracts/TellerV2Storage.sol [hunk 4]
```diff
@@ -138,4 +139,9 @@ abstract contract TellerV2Storage_G2 is TellerV2Storage_G1 {
...
+    mapping(uint256 => bool) _isBidCollateralBacked;
```
Comment (passabilities) [inline packages/contracts/contracts/TellerV2Storage.sol:144]: what are we using this for?

Comment (sypeer) [reply packages/contracts/contracts/TellerV2Storage.sol:144]: to check if a submitted bid has collateral before pulling in the funds from the borrower on bid acceptance

**Pull request closed without merging**